### PR TITLE
WEBITOOLS-106: When none provided, assigned a default exclusion rules and patterns

### PR DESCRIPTION
**Pull request by leon2Meudon92**
…and exclusion patterns as well

### CODE REVIEW

File: aip-console-tools-cli/README.md [hunk 1]
```diff
@@ -198,21 +198,37 @@ The available options are :
...
+  date and time.
 * `--verbose` (optional): Whether the command log should be output to the console or not, defaulted to true
-* `--exclude-patterns` or `-exclude`: File patterns to exclude in the delivery, the pattern needs to follow the syntax of [glob patterns](https://www.malikbrowne.com/blog/a-beginners-guide-glob-patterns)
-* `--auto-discover`: will discover new technologies and install new extensions during rescan, to disable when run consistency check
-* `--enable-security-dataflow` (optional): Enables the Security Dataflow objective for this version. <u>Has no impact when cloning a version</u>.
+* `--exclude-patterns` or `-exclude`: File patterns to exclude in the delivery, the pattern needs to follow the syntax
+  of [glob patterns](https://www.malikbrowne.com/blog/a-beginners-guide-glob-patterns)
+* `--exclusion-rules` or `-rules`: (Optional) case sensitive. Project's exclusion rules, separated with comma and no
+  space.</br>
+  *example* `--rules` = "EXCLUDE_EMPTY_PROJECTS,PREFER_FULL_DOT_NET_TO_BASIC_DOT_NET_WEB,EXCLUDE_TEST_CODE"
```
[the remark this answers is not in the document]
Comment (leon2Meudon92) [reply aip-console-tools-cli/README.md:212]: They are listed on command -help
Let me add this comment to help finding the list in case of

**Review by Krelix — CHANGES_REQUESTED** [review]
A few changes to be added

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java [hunk 2]
```diff
@@ -133,6 +137,9 @@ public class DeliverVersionCommand implements Callable<Integer> {
     @CommandLine.Option(names = {"-exclude", "--exclude-patterns"},
             description = "File patterns(glob pattern) to exclude in the delivery, separated with comma")
     private String exclusionPatterns;
+    @CommandLine.Option(names = {"-rules", "--exclusion-rules"}, split = ",", type = ExclusionRuleType.class
```
[the remark this answers is not in the document]
Comment (leon2Meudon92) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java:140]: updating

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java [hunk 3]
```diff
@@ -218,8 +225,11 @@ public Integer call() throws Exception {
 
             builder.objectives(VersionObjective.BLUEPRINT, blueprint);
             builder.objectives(VersionObjective.SECURITY, enableSecurityAssessment);
-
-            String deliveryConfigGuid = applicationService.createDeliveryConfiguration(applicationGuid, sourcePath, exclusionPatterns, cloneVersion);
+            Exclusions exclusions = Exclusions.builder().excludePatterns(exclusionPatterns).build();
+            if (exclusionRules != null && exclusionRules.length > 0) {
+                exclusions.setExclusionRules(Arrays.asList(exclusionRules).stream().collect(Collectors.toSet()));
```
[the remark this answers is not in the document]
Comment (leon2Meudon92) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java:230]: Set doesn't work. I tested it and still looking for a bean Set.

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/Exclusions.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@Setter
+@Builder
+public class Exclusions {
+    @Builder.Default
+    private String excludePatterns = null;
+    @Builder.Default
+    private Set<ExclusionRuleType> exclusionRules = ExclusionRuleType.getDefaultExclusionRules();
+
+    public static Set<String> getDefaultIgnorePatterns() {
+        return Arrays.asList("tmp/", "temp/", "*test", "tests", "target/", ".svn/", ".git/", "_Macosx/").stream().collect(Collectors.toSet());
```
[the remark this answers is not in the document]
Comment (leon2Meudon92) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/dto/Exclusions.java:21]: It's the way Console handle that pattern. I couldn't change it like that

File: aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Krelix) [reply aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/DeliverVersionCommand.java:230]: You didn't read my whole comment :)

Please replace Arrays.asList()... with the following (if it doesn't work, we can discuss this directly) :
```suggestion
                exclusions.setExclusionRules(Arrays.stream(exclusionRules).collect(Collectors.toSet()));
```

**Review by Krelix — APPROVED** [review]

**Pull request merged**